autogpt/autogpt/agents/agent.py

```diff
@@ -117,24 +121,29 @@ def __init__(
         self.system = SystemComponent()
         self.history = ActionHistoryComponent(
             settings.history,
-            self.send_token_limit,
             lambda x: self.llm_provider.count_tokens(x, self.llm.name),
-            legacy_config,
             llm_provider,
+            ActionHistoryConfiguration(
+                model_name=legacy_config.fast_llm, max_tokens=self.send_token_limit
+            ),
         ).run_after(WatchdogComponent)
-        self.user_interaction = UserInteractionComponent(legacy_config)
-        self.file_manager = FileManagerComponent(settings, file_storage)
+        self.user_interaction = UserInteractionComponent(
+            legacy_config.noninteractive_mode
+        )
+        self.file_manager = FileManagerComponent(file_storage, settings)
         self.code_executor = CodeExecutorComponent(
             self.file_manager.workspace,
-            settings,
-            legacy_config,
+            CodeExecutorConfiguration(
+                docker_container_name=f"{settings.agent_id}_sandbox"
+            ),
         )
-        self.git_ops = GitOperationsComponent(legacy_config)
-        self.image_gen = ImageGeneratorComponent(
-            self.file_manager.workspace, legacy_config
+        self.git_ops = GitOperationsComponent()
+        self.image_gen = ImageGeneratorComponent(self.file_manager.workspace)
+        self.web_search = WebSearchComponent()
+        self.web_selenium = WebSeleniumComponent(
+            llm_provider,
+            legacy_config.app_data_dir,
         )
```

Comment:
Does this just continue to grow as we add more components?

Reply:
Yes, at some point we might want to have some component management system but that's a problem for the future.